src/SignalR/server/Core/src/HubConnectionHandler.cs

```diff
@@ -200,6 +200,9 @@ private async Task HubOnDisconnectedAsync(HubConnectionContext connection, Excep
         // Ensure the connection is aborted before firing disconnect
         await connection.AbortAsync();
 
+        // If a client result is requested in OnDisconnectedAsync we want to make sure it isn't blocked by the ActiveInvocationLimit
+        _ = connection.ActiveInvocationLimit.AttemptWait();
```

Comment:
I don't understand this. If a caller client result is requested in OnDisconnectedAsync, it will fail with a client disconnected error, right? It seems like this is trying to make sure the invocation doesn't block before that error. But if that's the case, wouldn't we want to call Release?

Reply:
> If a caller client result is requested in OnDisconnectedAsync, it will fail with a client disconnected error, right?

Right.

> It seems like this is trying to make sure the invocation doesn't block before that error. But if that's the case, wouldn't we want to call Release?

`InvokeAsync` calls Release, so we need to Acquire here in order to make the Release not block.